src/main/java/com/kobylynskyi/graphql/codegen/mapper/RequestResponseDefinitionToDataModelMapper.java

```diff
@@ -154,13 +157,19 @@ private static ProjectionParameterDefinition getTypeNameProjectionParameterDefin
         String className = DataModelMapper.getParametrizedInputClassName(mappingContext, fieldDefinition, parentTypeDefinition);
         Map<String, Object> dataModel = new HashMap<>();
         // ParametrizedInput classes are sharing the package with the model classes, so no imports are needed
+        List<ParameterDefinition> fields = inputValueDefinitionToParameterMapper.map(
+                mappingContext, fieldDefinition.getInputValueDefinitions(), parentTypeDefinition.getName());
+        //kotlin not support empty
+        if (mappingContext.getGeneratedLanguage().equals(GeneratedLanguage.KOTLIN) &&
```

Comment:
Kotlin does support classes with no fields, we just need to use regular classes (not data classes) for them.
This is because equals, hashcode etc don't make sense if there are no fields.